x-pack/plugins/observability_solution/slo/public/hooks/use_permissions.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { sloFeatureId } from '@kbn/observability-plugin/common';
+import { useKibana } from '../utils/kibana_react';
+import { useFetchSloGlobalDiagnosis } from './use_fetch_global_diagnosis';
+
+export function usePermissions() {
```

Comment:
💯 Can we add hook tests?

Reply:
Even better than hook tests would be a functional test with real test users that confirms we see errors for restricted users and the SLO page for allowed users.. @mgiota got the functional test ball rolling.

Reply:
Added the tests https://github.com/elastic/kibana/pull/182609/commits/5eb629c28f3014dccd02bd62de492a1df970bba0